node/node.go

```diff
@@ -153,24 +153,17 @@ func (n *Node) Start() error {
 
 	// Initialize the p2p server. This creates the node key and
 	// discovery databases.
-	n.serverConfig = p2p.Config{
-		PrivateKey:       n.config.NodeKey(),
-		Name:             n.config.NodeName(),
-		Discovery:        !n.config.NoDiscovery,
-		DiscoveryV5:      n.config.DiscoveryV5,
-		DiscoveryV5Addr:  n.config.DiscoveryV5Addr,
-		BootstrapNodes:   n.config.BootstrapNodes,
-		BootstrapNodesV5: n.config.BootstrapNodesV5,
-		StaticNodes:      n.config.StaticNodes(),
-		TrustedNodes:     n.config.TrusterNodes(),
-		NodeDatabase:     n.config.NodeDB(),
-		ListenAddr:       n.config.ListenAddr,
-		NetRestrict:      n.config.NetRestrict,
-		NAT:              n.config.NAT,
-		Dialer:           n.config.Dialer,
-		NoDial:           n.config.NoDial,
-		MaxPeers:         n.config.MaxPeers,
-		MaxPendingPeers:  n.config.MaxPendingPeers,
+	n.serverConfig = n.config.P2P
+	n.serverConfig.PrivateKey = n.config.NodeKey()
+	n.serverConfig.Name = n.config.NodeName()
+	if n.serverConfig.StaticNodes == nil {
+		n.serverConfig.StaticNodes = n.config.StaticNodes()
+	}
+	if n.serverConfig.StaticNodes == nil {
+		n.serverConfig.TrustedNodes = n.config.TrusterNodes()
+	}
+	if n.serverConfig.NodeDatabase == "" {
+		n.serverConfig.NodeDatabase = n.config.NodeDB()
```

Comment:
When does it happen that `n.serverConfig.Xyz` is not nil? I.e. when the above code should not override?